src/biome/text/pipelines/pipeline.py

```diff
@@ -272,6 +273,33 @@ def predict_json(self, inputs: JsonDict) -> Optional[JsonDict]:
         output = self.model.forward_on_instance(instance)
         return sanitize(output)
 
+    def predict_json_with_cache(self, inputs: JsonDict) -> Optional[JsonDict]:
+        """Predict an input with the pipeline's model while caching the result.
+
+        For this we need to pickle the input dict so we can hash it.
+
+        Parameters
+        ----------
+        inputs
+            The input features/tokens in form of a json dict
+
+        Returns
+        -------
+        output
+            The model's prediction in form of a dict.
+            Returns None if the input could not be transformed to an instance.
+        """
+        pickled_inputs = pickle.dumps(inputs)
```

Comment:
Could we create a hashable dict available for cache parameter hashing? I have the impression that it is much more efficient than pickle/unpickle the dictionary 

```python
class Hashabledict(dict):
    def __hash__(self):
        return hash(frozenset(self))

hashable_inputs = Hashabledict(inputs)
predict(hashable_inputs)
```